rancher/charts
# [dev-v2.8] Bump windows fluent agent version to 2.2.0

## Issue: https://github.com/rancher/rancher/issues/45308

## Problem
A prior PR bumped the fluent bit agent version for windows to `v1.8.10`, in order to remove foreign layers. However It missed the fact that the logging operator chart version was upgraded to `4.4.0`. `4.4.0` introduced the `Kube_Token_TTL` field and the `v1.8.10` version of fluent bit agent on windows does not understand this value, resulting in a crash loop

## Solution
Bump the fluent bit agent version used by windows to `2.2.0` 

## Testing
I have performed the following tests
+ Install `v103.1.0+up4.4.0` onto a windows cluster, confirmed that the windows agent does not deploy successfully 
+ Upgrade to `v103.1.1-rc1+up4.4.0`
+ Confirmed that the windows agent successfully upgraded from `rancher/fluent-bit:v1.8.10` to `rancher/fluent-bit:2.2.0`
+ Create a simple file log output
+ Create a simple flow which points to a dummy windows deployment 
+ Create a dummy windows deployment which logs `OK` every second
+ Shell into the fluentd container and confirm that the logs are present and in the correct path specified by the Output

<details>
<summary>
Test Output
</summary>

```
apiVersion: logging.banzaicloud.io/v1beta1
kind: Output
metadata:
  name: fileoutput
  namespace: default
spec:
  file:
    path: /tmp/${tag}
```
</details>

<details>
<summary>
Test Workload
</summary>

```
  apiVersion: apps/v1
  kind: Deployment
  metadata:
    labels:
      app: win-webserver
    name: win-webserver
    namespace: default
  spec:
    replicas: 1
    selector:
      matchLabels:
        app: win-webserver
    template:
      metadata:
        labels:
          app: win-webserver
        name: win-webserver
      spec:
        containers:
          - name: windowswebserver
            image: mcr.microsoft.com/windows/servercore:ltsc2022
            command:
              - powershell.exe
              - -command
              - "while(1){Write-Host OK; sleep 1;}"
        nodeSelector:
          kubernetes.io/os: windows
```
</details>

## Engineering Testing
### Manual Testing


### Automated Testing


## QA Testing Considerations


### Regressions Considerations


## Backporting considerations
This same change will be forward ported to 2.9. This issue does not reproduce on 2.7 as the logging operator has not been bumped.
